src/Datadog.Trace/Agent/Transports/HttpClientRequest.cs

```diff
@@ -22,10 +22,10 @@ public void AddHeader(string name, string value)
             _request.Headers.Add(name, value);
         }
 
-        public async Task<IApiResponse> PostAsync(Span[][] traces, FormatterResolverWrapper formatterResolver)
+        public async Task<IApiResponse> PostAsync(ArraySegment<byte> traces)
         {
             // re-create HttpContent on every retry because some versions of HttpClient always dispose of it, so we can't reuse.
-            using (var content = new TracesMessagePackContent(traces, formatterResolver))
+            using (var content = new BinaryTracesMessagePackContent(traces))
```

Comment:
Can we use the built-in `ByteArrayContent` now instead of a custom one?
```suggestion
            using (var content = new ByteArrayContent(traces.Array, traces.Offset, traces.Count))
```